test/stateful/eventSourceConstruct.test.ts

```diff
@@ -0,0 +1,121 @@
+import * as cdk from 'aws-cdk-lib';
+import { Match, Template } from 'aws-cdk-lib/assertions';
+import { EventSource } from '../../lib/workload/stateful/event_source/component';
+
+let stack: cdk.Stack;
+
+function assert_common(template: Template) {
+  template.resourceCountIs('AWS::SQS::Queue', 2);
+
+  template.hasResourceProperties('AWS::SQS::Queue', {
+    QueueName: 'queue',
+    RedrivePolicy: {
+      deadLetterTargetArn: Match.anyValue(),
+      maxReceiveCount: 100,
+    },
+  });
+
+  template.hasResourceProperties('AWS::CloudWatch::Alarm', {
+    ComparisonOperator: 'GreaterThanThreshold',
+    EvaluationPeriods: 1,
+    Threshold: 0,
+  });
+
+  template.hasResourceProperties('AWS::Events::Rule', {
+    EventPattern: {
+      source: ['aws.s3'],
+      detail: {
+        bucket: {
+          name: 'bucket',
+        },
+      },
+    },
+  });
+}
+
+beforeEach(() => {
+  stack = new cdk.Stack();
+});
+
+test('Test EventSource created props', () => {
+  new EventSource(stack, 'TestDatabaseConstruct', {
```

Comment:
I guess `TestDatabaseConstruct` should be updated to `TestEventSourceConstruct`... 
(also for the following tests)